noir_stdlib/src/array/mod.nr

```diff
@@ -114,25 +66,39 @@ impl<T, let N: u32> [T; N] {
     }
 }
 
+impl<T, let N: u32> [T; N] where T: Ord + Eq {
+    pub fn sort(self) -> Self {
+        self.sort_via(|a: T, b: T| a <= b)
+    }
+}
+
+impl<T, let N: u32> [T; N] where T: Eq {
+
+    pub fn sort_via<Env>(self, ordering: fn[Env](T, T) -> bool) -> Self {
+        unsafe {
+            // Safety: `sorted` array is checked to be:
+            //   a. a permutation of `input`'s elements
+            //   b. satisfying the predicate `ordering`
+            let sorted = quicksort::quicksort(self, ordering);
+
+            if !is_unconstrained() {
+                for i in 0..N - 1 {
+                    assert(ordering(sorted[i], sorted[i + 1]));
+                }
```

Comment:
Looks like we'd still be failing if arrays contained any equal elements: https://github.com/noir-lang/noir/issues/5780

Before it was just a sanity check but now if we're relying on this to constrain the sorted outputs then maybe we need documentation that the sort function must be a `<=` instead of a `<`?

Reply:
I've added some documentation for this issue with duplicate elements.